internal/services/network/network_manager_routing_rule_resource.go

```diff
@@ -0,0 +1,341 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.ResourceWithUpdate        = ManagerRoutingRuleResource{}
+	_ sdk.ResourceWithCustomizeDiff = ManagerRoutingRuleResource{}
+)
+
+type ManagerRoutingRuleResource struct{}
+
+func (ManagerRoutingRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return routingrules.ValidateRuleID
+}
+
+func (ManagerRoutingRuleResource) ResourceType() string {
+	return "azurerm_network_manager_routing_rule"
+}
+
+func (ManagerRoutingRuleResource) ModelObject() interface{} {
+	return &ManagerRoutingRuleResourceModel{}
+}
+
+type ManagerRoutingRuleResourceModel struct {
+	Description      string                          `tfschema:"description"`
+	Destination      []ManagerRoutingRuleDestination `tfschema:"destination"`
+	Name             string                          `tfschema:"name"`
+	NextHop          []ManagerRoutingRuleNextHop     `tfschema:"next_hop"`
+	RuleCollectionId string                          `tfschema:"rule_collection_id"`
+}
+
+type ManagerRoutingRuleDestination struct {
+	Address string `tfschema:"address"`
+	Type    string `tfschema:"type"`
+}
+
+type ManagerRoutingRuleNextHop struct {
+	Address string `tfschema:"address"`
+	Type    string `tfschema:"type"`
+}
+
+func (ManagerRoutingRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9\_\.\-]{1,64}$`),
+				"`name` must be between 1 and 64 characters long and can only contain letters, numbers, underscores(_), periods(.), and hyphens(-).",
+			),
+		},
+
+		"rule_collection_id": commonschema.ResourceIDReferenceRequiredForceNew(&routingrules.RuleCollectionId{}),
+
+		"destination": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"address": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(routingrules.PossibleValuesForRoutingRuleDestinationType(), false),
+					},
+				},
+			},
+		},
+
+		"next_hop": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(routingrules.PossibleValuesForRoutingRuleNextHopType(), false),
+					},
+					"address": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Same here, shouldn't this be using the `validation.IsCIDR` and/or `validation.IsIPAddress` functions to verify this instead of just checking for and empty string here?

Reply:
updated, now it valid ip address is required